backend/src/dao/controller/member.controller.ts

```diff
@@ -0,0 +1,83 @@
+import { Body, Controller, Delete, Get, HttpCode, Param, Post, Put, UseGuards } from '@nestjs/common'
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger'
+import { Auth } from 'src/auth/auth.decorator'
+import { RolesGuard } from 'src/auth/auth.roles.guard'
+import { User } from 'src/auth/user.decorator'
+import { DaoAuth } from 'src/dao/dao.auth.decorator'
+import { Role } from 'src/user/user.dto'
+import { CreateMemberDto, ExtendedMemberDto, MemberDto, MemberInviteDto, MemberQueryDto } from '../../member/member.dto'
+import { MemberService } from '../../member/member.service'
+
+@ApiBearerAuth()
+@ApiTags('member')
+@Controller('/dao/:daoId/member')
+export class MemberController {
+  constructor(private memberService: MemberService) {}
+
+  @Post('list')
+  @HttpCode(200)
+  @DaoAuth(Role.founder, Role.projectManager)
+  list(@Param('daoId') daoId: string, @Body() query: MemberQueryDto): Promise<ExtendedMemberDto[]> {
+    query.daoId = daoId
+    return this.memberService.list(query)
+  }
+
+  @Post('invite')
+  @HttpCode(200)
+  @DaoAuth(Role.founder, Role.projectManager)
+  invite(
+    @Param('daoId') daoId: string,
+    @User('userId') userId: string,
+    @Body() invite: MemberInviteDto,
+  ): Promise<MemberDto> {
+    invite.daoId = invite.daoId
```

Comment:
Help me unpack the intention of this method.

Someone with Founder or Project Manager role sends an invite to a member to do what?

Is the intention here for a project manager to invite a contributor to join a project or take on a task?

Let's clarify that an invitation to participate in a project does not yet mean that the contributor will receive sweat tokens. That happens only when they produce a deliverable and after project manager approval, they trade it for sweat tokens.

Also a contributor may already have sweat tokens from previous projects they participated in. Or they may be a brand new contributor with no sweat tokens granted yet.

My point is that we need to distinguish between participation in a project and owning sweat tokens.

Reply:
This specific api call just create a member record to track user activities and role inside the DAO, no other meaning so far.
Let's say you own a DAO and invite me as PM. I then lead a project and in turn I invite other users to carry on task. 

So far there are no implicit advantages for users, see it just as a way to assign roles inside a DAO

Reply:
> This informal offchain participation probably does not need to be represented in the SporosDAO app in the MPV.

I agree but how do you internally track roles and entity ownership? Eg. User A own a Project, User B is contributing to task X ?

> Handle sweat token projects with multi-week, multi contributor scope. Sweat tokens are non-transferrable.

That's ok, maybe you have different expectations on the implementation approach?

So far I am following a web2-ish approach to provide the base functionalities as per above statement while we will integrate to KaliDAO once the PM/contributor flow is working (I would say task assignment, reporting and task confirmation).

The KaliDAO integration comes later because we store on-chain only the relevant parts (e.g. DAO creation, project creation and assignment, dispute resolution, Sweat Token conversion). 

To recap, my take on this is that the on-chain integration (the legal framework gateway) hooks on a sub-layer that has a few known steps but does not have specific assumptions on how it should work internally. Before jumping into smart contract dev I would see the basic flow validated and improved enough to be reasonably stable. 

What do you think ?

> I imagine we would have a proposal type named SweatEquityProjectProposal with these parameters:

That is a very good hint, if the above points are good for you I would move to (an expanded version of) #4

Reply:
Solid!